Add a getting started guide to documentation

decentral1se opened the pull request:
I think this is an OK start.

#### PR Type
- Docs Pull Request

decentral1se commented:
/cc @samccann @acozine

webknjaz commented on README.rst:
```diff
@@ -25,106 +25,49 @@ Molecule
    :alt: Repository License
 
-Molecule is designed to aid in the development and testing of `Ansible`_ roles.
+Molecule is designed to aid in the development and testing of Ansible roles.
```
Can we please keep the reference?
```suggestion
Molecule is designed to aid in the development and testing of `Ansible`_ roles.
```

webknjaz commented on doc/source/getting-started.rst:
```diff
@@ -0,0 +1,194 @@
+.. note::
+
+    In order to complete this guide by hand, you will need to additionally
+    install `Docker`_.
```
I know that @acozine was having troubles with the initial setup under macOS and as a result, she wanted to add at least these details do docs:
- `pip install molecule[docker]` for also installing appropriate python client library seamlessly;
- `docker run hello-world` as advice on how to verify that docker itself is set up correctly.

webknjaz commented on doc/source/getting-started.rst:
```diff
@@ -0,0 +1,194 @@
+not, please review the `Role directory structure`_ guide to see what each
+folder is responsible for.
+
+.. _Ansible Galaxy: https://docs.ansible.com/ansible/latest/reference_appendices/galaxy.html#the-command-line-tool
```
Could you please try this?
```suggestion
.. _Ansible Galaxy: https://docs.ansible.com/ansible/latest/reference_appendices\
                    /galaxy.html#the-command-line-tool
```
(I didn't test it, though. I hope to make links fit into the common line-length limit)
But it's minor, disregard it in any event of failure or if you think that the link shouldn't be broken into pieces.

webknjaz commented on doc/source/getting-started.rst:
```diff
@@ -0,0 +1,194 @@
+folder is responsible for.
+
+.. _Ansible Galaxy: https://docs.ansible.com/ansible/latest/reference_appendices/galaxy.html#the-command-line-tool
+.. _Role directory structure: https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html#role-directory-structure
```
I'm curious whether it's possible to reference this using intersphinx. I think it's more flexible this way.

decentral1se replied:
Hooked up InterSphinx! Much better, thanks for recommending.

webknjaz commented on doc/source/getting-started.rst:
```diff
@@ -0,0 +1,194 @@
+  state of the container after your role has finished executing. Other verifier
+  tools are available.
+
+.. _Docker: https://docs.docker.com/
```
This reference is declared 3 times in this document. I think one should be enough for it to work.

decentral1se commented:
Thanks for the review @webknjaz. Addressed those comments.

I'm using ``--fixup`` to make my incremental changes easier to review. Will rebase once we're done.

webknjaz commented on README.rst:
```diff
@@ -26,105 +26,48 @@ Molecule
-.. _`playbooks`: https://docs.ansible.com/ansible/latest/playbooks.html
-.. _`role`: https://docs.ansible.com/ansible/latest/playbooks_roles.html
-
-Quick Start
```
@decentral1se do you intend to completely remove this from README (which would include GitHub repo view and PyPI page)?

There's a way to keep this in one source file (README) and then include parts of readme into different pages on RTD using `include` SphinX directive with `start-after`/`end-before` markers.

Example: https://github.com/ansible/ansible-lint/pull/420/files#diff-3caecf318f72e748f256c37866cc61f4R13

decentral1se replied:
Awesome! That was much needed, I rigged it up in https://github.com/ansible/molecule/pull/1776/commits/d740306d5302be04b3d4711d8df4afce4e531d98.

webknjaz commented on doc/source/conf.py:
```diff
@@ -77,10 +77,12 @@
 
 intersphinx_mapping = {
-    'ansible': ('https://docs.ansible.com/ansible/devel/', None),
+    'ansible': ('https://docs.ansible.com/ansible/latest/', None),
```
FTR: Unlike commonly known practice, `latest` in Ansible docs corresponds to the stable version, not to the latest state of the main Git branch. Is this your intention? Just checking...

decentral1se replied:
Yep, stable seems to be more suitable place to link to.

gundalow commented on README.rst:
```diff
@@ -25,106 +25,61 @@ Molecule
+Get Involved
+============
 
 * Join us in the ``#ansible-molecule`` channel on `freenode`_.
```
```suggestion
* Join us in the ``#ansible-molecule`` channel on `Freenode`_.
```

gundalow commented on README.rst:
```diff
@@ -25,106 +25,61 @@ Molecule
+* Want to know about releases, subscribe to `ansible-announce list`_.
+* For the full list of Ansible email Lists, IRC channels see the `communication page`_.
 
 .. _`freenode`: https://freenode.net
```
```suggestion
.. _`Freenode`: https://freenode.net
```

gundalow commented on doc/source/getting-started.rst:
```diff
@@ -0,0 +1,196 @@
+    'molecule[docker]'``. Otherwise, you can ``pip install docker``.
+
+.. _Docker: https://docs.docker.com/
+
```
I wonder if a bullet point overview of the rest of the page here would help, give some content for what people will read as well as giving high level context/

decentral1se replied:
Solid idea! `toctree` to the rescue ...

samccann commented on doc/source/getting-started.rst:
```diff
@@ -0,0 +1,196 @@
+
+.. note::
+
+    For future reference, if you want to initialise Molecule within an existing
```
```suggestion
    For future reference, if you want to initialize Molecule within an existing
```

samccann commented on doc/source/getting-started.rst:
```diff
@@ -0,0 +1,196 @@
+In this folder is a single :ref:`root_scenario` called ``default``.
+
+Scenarios are the starting point for a lot of powerful functionality that
+Molecule offers. For now, we can think of a Scenario as a test suite for your
```
```suggestion
Molecule offers. For now, we can think of a scenario as a test suite for your
```

samccann commented on doc/source/getting-started.rst:
```diff
@@ -0,0 +1,196 @@
+* The :ref:`provisioner`. Molecule only provides an Ansible provisioner.
+  Ansible manages the life cycle of the instance based on this configuration.
+
+* The :ref:`root_scenario` defintion. Molecule relies on this configuration to
```
```suggestion
* The :ref:`root_scenario` definition. Molecule relies on this configuration to
```

decentral1se commented on doc/source/getting-started.rst:
```diff
@@ -0,0 +1,196 @@
+    In order to complete this guide by hand, you will need to additionally
+    install `Docker`_. Molecule requires an external Python dependency for the
+    Docker driver which is provided if you install Molecule using ``pip install
+    'molecule[docker]'``. Otherwise, you can ``pip install docker``.
```
Yeah, that makes sense! `molecule[docker]` will do for now :+1:

decentral1se commented:
Thanks @gundalow @samccann! Much appreciated. Comments addressed :+1:

webknjaz commented:
@decentral1se (this is for another PR but) we could set up spell checker in CI. Example: https://github.com/aio-libs/aiohttp/blob/398b5ba/docs/Makefile#L179-L182 + `apt install libenchant-dev` + `pip install sphinxcontrib-spelling` + it supports adding terms to the whitelist

acozine commented on CONTRIBUTING.rst:
```diff
@@ -132,3 +132,36 @@ To bring in updated upstream modules.  Update `gilt.yml` and execute the followi
+Documentation
+=============
+
+Working with InterSphinx
```
Why are we adding a generic example of using InterSphinx to the molecule docs?

decentral1se replied:
Well, we'll want contributors to work with InterSphinx and I didn't see such a step-by-step on how to get the links out of the inventory listing and how to read the link syntax (so I wouldn't think this is *that* generic), so I put that here. Sphinx can be intimidating to work with and I'd rather have the 'this is all you need to know' docs here locally in Molecule. That's my two cents!

webknjaz replied:
That's true. It's quite hard to find a comprehensive guide on how to use intersphinx. Sometimes I have to google for it but I only find mentions of half-baked commands here and there.
So it's probably reasonable to keep it here until sphinx starts having their own instructions...

acozine replied:
Okay. If I find a better solution, I'll propose it in a future PR.

acozine commented:
This is an improvement over what we have now - thanks @decentral1se.

decentral1se commented:
OK, I've squashed all my commits, should be good to go :+1: 

Thanks for reviews folks.

webknjaz commented on README.rst:
```diff
@@ -25,106 +25,61 @@ Molecule
 
-Molecule is designed to aid in the development and testing of `Ansible`_ roles.
+Molecule is designed to aid in the development and testing of
+:std:doc:`Ansible <ansible:index>` roles.
```
@decentral1se late notice: you cannot do this in README because it's included in PyPI.

* https://travis-ci.com/ansible/molecule/jobs/180975740#L238
* https://travis-ci.com/ansible/molecule/jobs/180975741#L238

gundalow replied:
Fixed in `master`
